elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -140,24 +142,33 @@ export class PeerPool extends EventEmitter {
 
 		// This needs to be an arrow function so that it can be used as a listener.
 		this._handleDiscoverPeer = (peerInfo: P2PDiscoveredPeerInfo) => {
+			const peerId = constructPeerIdFromPeerInfo(peerInfo);
+			// Add a peer to a list that are successfully connected
+			this._connectedPeers.set(peerId, peerInfo);
+
 			// Re-emit the message to allow it to bubble up the class hierarchy.
 			this.emit(EVENT_DISCOVERED_PEER, peerInfo);
 		};
 
 		this._handlePeerConnect = async (peerInfo: P2PDiscoveredPeerInfo) => {
+			const peerId = constructPeerIdFromPeerInfo(peerInfo);
+			// Add a peer to a list that are successfully connected
+			this._connectedPeers.set(peerId, peerInfo);
+
 			// Re-emit the message to allow it to bubble up the class hierarchy.
 			this.emit(EVENT_CONNECT_OUTBOUND, peerInfo);
```

Comment:
Seems redundant as this event is fired just before `EVENT_DISCOVERED_PEER` or am I wrong here?

Reply:
You are right it's not needed

Reply:
True. Besides, we have to check if the list has that peer before setting it.

Reply:
@diego-G we don't have to check explicitly if the peerId exists or not because when we set, if it exists then it will update the value else it will create a new entry for it

Reply:
@ishantiw would you address @4miners suggestion in this PR?

Reply:
This PR is not adding new code but fixing the existing code, another thing is that PeerPool doesn't have any unit tests, most of its functionality is covered in integration tests. If I write unit tests for PeerPool then I will have to cover all the tests which will take the time or maybe I can just write unit tests for the functions? Another thing is, a lot of these unit tests will not be there or will change in 2.2 version. That was the reason that @jondubois recommended to have low priority on unit tests

Reply:
@ishantiw Yes, you can cover by unit tests the new or modified code only, the rest can be done in a separate issue.